Encoder/JWTEncoderInterface.php

```diff
@@ -14,21 +14,23 @@
 {
     /**
      * @param array $data
+     * @param array $header
      *
      * @return string the encoded token string
      *
      * @throws JWTEncodeFailureException If an error occurred while trying to create
      *                                   the token (invalid crypto key, invalid payload...)
      */
-    public function encode(array $data);
+    public function encode(array $data, array $header = []);
 
     /**
      * @param string $token
+     * @param array  $header
      *
      * @return array
      *
      * @throws JWTDecodeFailureException If an error occurred while trying to load the token
      *                                   (invalid signature, invalid crypto key, expired token...)
      */
-    public function decode($token);
+    public function decode($token, array &$header = []);
```

Comment:
Actually I would not change `decode()` as we don't allow to pass a payload here, just an immutable (stringish) token, its content should not be altered at this stage I think. But maybe you have a specific use case in mind?